encord/utilities/benchmark_utilities.py

```diff
@@ -0,0 +1,210 @@
+                similarities_found = []  # Store tuples following (which_dict_the_object_came_from, object_index) format
+                for comparing_index, comparing_dict in enumerate(object_dicts):
+                    if base_index == comparing_index:  # skip comparison between same agent's annotations
+                        continue
+                    max_similarity = (-1, -1)  # (best_jaccard_similarity_coefficient, index_in_comparing_dict)
+                    for idx2, obj2 in comparing_dict.items():
+                        cur_similarity = (__calculate_jaccard_similarity(obj1, obj2), idx2)
+                        max_similarity = max(max_similarity, cur_similarity)
+                    if max_similarity[0] >= threshold:
+                        similarities_found.append((comparing_index, max_similarity[1]))
+
+                # Check if at least half of the agents agreed on obj1 and if so then delete those objects similar to
+                # obj1 in order to avoid choosing other representation of the same object in a later iteration
+                if (len(similarities_found) + 1) * 2 >= amount_agents:
+                    objects_used_in_base_dict.append(idx1)
+                    # Use dictionary so item deletion is O(1) in the average case instead of O(n)
+                    object_dict = consensus_objects.setdefault(feature_hash, dict())
+                    object_dict[len(object_dict)] = obj1
+                    for dict_index, object_index in similarities_found:
+                        object_dicts[dict_index].pop(object_index)
+
+            # Delete current agent's agreed by majority objects to avoid deleting entries in the dict while iterating it
+            for object_index in objects_used_in_base_dict:
+                object_dicts[base_index].pop(object_index)
+    return consensus_objects
+
+
+def __calculate_jaccard_similarity(obj1, obj2):
+    """
+    Calculate Jaccard similarity coefficient (Intersection over Union) between objects obj1 and obj2.
+    """
+    p1 = __transform_obj_to_polygon(obj1)
+    p2 = __transform_obj_to_polygon(obj2)
+    intersection = p1.intersection(p2).area
+    union = p1.area + p2.area - intersection
+    return intersection / union
+
+
+def __transform_obj_to_polygon(obj):
+    """
+    Return the polygon represented in obj.
+    """
+    if obj["shape"] == "bounding_box":
+        x = obj["boundingBox"]["x"]
+        y = obj["boundingBox"]["y"]
+        h = obj["boundingBox"]["h"]
+        w = obj["boundingBox"]["w"]
+        return box(x, y, x + w, y + h)
+    elif obj["shape"] == "polygon":
+        poly_dict = obj["polygon"]
+        return Polygon([(poly_dict[str(i)]["x"], poly_dict[str(i)]["y"]) for i in range(len(poly_dict))])
+    elif obj["shape"] == "point":
+        raise NotImplementedError("Skeleton type is not fully functional so any comparison involving it is not allowed")
```

Comment:
That message doesn't look right.

Reply:
Yes, Eric told to raise an exception if someone tried to use points because skeleton (points) logic is not available/ready for now and as I don't know the underlying structure of this type there is no much to do.

Reply:
Chipping in on this one -- points are actually separate from skeletons. Point is a keypoint. A skeleton would be a skeleton shape.

Reply:
Yes, Denis showed me some examples using skeleton as a shape which I wasn't aware at the time that it is possible and I fixed the code to fulfill the restrictions.